### PR TITLE
Add Database Migration Service sdk for public preview

### PR DESCRIPTION
**Pull request by hitenjava**
Link to swagger here - 
https://github.com/Azure/azure-rest-api-specs/tree/current/specification/datamigration/resource-manager/Microsoft.DataMigration/2017-11-15-preview

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <Description>DataMigration.Tests Class Library</Description>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.KeyVault" Version="2.3.2" />
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="15.3.0" />
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: @hitenjava Microsoft.NET.Test.Sdk and Rest.ClientRuntime.Azure.TestFramework references are implicitly added to your project, so you don't have to explicitlty add them.
When we change these packages or upgrade them, your project will be automatically upgraded at the same time.
Having such explicit references inhibit us to upgrade repo wide such changes.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Everything looks good. Good to merge after addressing minor feedback given.

**Comment by rohit-joy** [comment]
@hitenjava Update the full Azure github path to the swagger in the summary: https://github.com/Azure/azure-rest-api-specs/tree/current/specification/datamigration/resource-manager/Microsoft.DataMigration/2017-11-15-preview

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj [hunk 2]
```diff
@@ -13,8 +13,6 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.KeyVault" Version="2.3.2" />
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
-    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="15.3.0" />
```
Comment (rohit-joy) [inline src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: @hitenjava Does the project build after removing the two lines? Did you remove them using the Nuget Package Manager? There may still be references to the binaries in the csproj.

**Review by rohit-joy — APPROVED** [review]

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj (hunk 1 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: Fixed.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**